src/core/src/pass/serialize.cpp

```diff
@@ -1016,7 +1016,10 @@ void ngfunction_2_ir(pugi::xml_node& netXml,
                     auto& rt_attribute = item.second.as<ov::RuntimeAttribute>();
                     const auto& type_info = rt_attribute.get_type_info();
                     attribute_node.append_attribute("name").set_value(type_info.name);
-                    attribute_node.append_attribute("version").set_value(type_info.get_version().c_str());
+                    // OpenVINO 2024.4 and older require "version" attribute to be present. It's safe to store fixed
+                    // values as all deserialized RuntimeAttribute devired attributes' version is 0 up to 2024.4
+                    assert(type_info.get_version().empty());
```

Comment:
is it possible to use `static_assert` here? I believe info about the version have to be available in compile time
if not, should we use OV assert here with some "error" message?